core/src/main/java/com/amplifyframework/auth/AuthCategoryBehavior.java

```diff
@@ -154,4 +155,28 @@ void signInWithFacebook(
     void fetchAuthSession(
             @NonNull Consumer<AuthSession> onSuccess,
             @NonNull Consumer<AuthException> onError);
+
+    /**
+     * Trigger password recovery for the given username.
+     * @param username This can be a typical username, email, phone number, etc. depending on the configuration
+     * @param onSuccess Success callback
+     * @param onError Error callback
+     */
+    void forgotPassword(
+            @NonNull String username,
+            @NonNull Consumer<AuthCodeDeliveryDetails> onSuccess,
+            @NonNull Consumer<AuthException> onError);
+
+    /**
+     * Complete password recovery process by inputting user's desired new password and confirmation code.
```

Comment:
`and the confirmation code that came from (where does it come from)`?

Reply:
It depends how it's configured so I think I'll leave it as is.